mbed/mbed.py

```diff
@@ -2221,6 +2231,25 @@ def compile_(toolchain=None, target=None, profile=False, compile_library=False,
                   + (['-v'] if verbose else [])
                   + args,
                   env=env)
+ 
+            if flash:
+                fw_name = artifact_name if artifact_name else program.name
+                fw_fbase = os.path.join(build_path, fw_name)
+                fw_file = fw_fbase + ('.hex' if os.path.exists(fw_fbase+'.hex') else '.bin')
+                if not os.path.exists(fw_file):
+                    error("Build program file (firmware) not found \"%s\"" % fw_file, 1)
+                detected = program.detect_target()
+
+                try:
+                    from mbed_host_tests.host_tests_toolbox import flash_dev, reset_dev
+                except (IOError, ImportError, OSError):
+                    error("The '-f/--flash' option requires that the 'mbed-greentea' python module is installed.\nYou can install mbed-ls by running 'pip install mbed-greentea'.", 1)
+
+                if not flash_dev(detected['msd'], fw_file, program_cycle_s=2):
+                    error("Unable to flash the target board connected to your system.", 1)
```

Comment:
Why not have that check that in htrun instead, and have the top function handle it?